### PR TITLE
testament: make valgrind optional

### PR DESCRIPTION
**Pull request by CyberTailor**
Add a simple switch to enable/disable valgrind support. Off by default

### CODE REVIEW

**Comment by Araq** [comment]
What's the point? testament already supports valgrind.

File: testament/testament.nim [hunk 1]
```diff
@@ -487,7 +489,7 @@ proc testSpecHelper(r: var TResults, test: var TTest, expected: TSpec,
             args = @["--unhandled-rejections=strict", exeFile] & args
           else:
             exeCmd = exeFile.dup(normalizeExe)
-            if expected.useValgrind != disabled:
+            if useValgrind and expected.useValgrind != disabled:
```
Comment (Araq) [inline testament/testament.nim]: Shouldn't this be `useValgrind or ...`?

Comment (CyberTailor) [reply testament/testament.nim]: `useValgrind` is a global variable. Probably should be renamed to avoid confusion

**Comment by CyberTailor** [comment]
> What's the point? testament already supports valgrind.

New option is for cases when running Valgrind is not desired (e.g. architecture or instruction set is not supported by Valgrind).

**Review by Varriount — APPROVED** [review]

**Pull request merged**